chore(release): add release notes for new operator versions

dogonthehorizon opened the pull request:
<!-- If Jira, enclose the ticket in brackets. Jira-GitHub integration requires this format. -->
Resolves Jira: [BOB-30677]

Includes:
  - Armory Operator 1.7.0
  - Spinnaker Operator 1.3.0

Also adds a support matrix include for Operator versions. We should
definitely include that in a message somewhere in the 2.28 notes when we
get there.

[BOB-30677]: https://armory.atlassian.net/browse/BOB-30677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dogonthehorizon committed:
chore(release): add release notes for new operator versions

Includes:
  - Armory Operator 1.7.0
  - Spinnaker Operator 1.3.0

Also adds a support matrix include for Operator versions. We should
definitely include that in a message somewhere in the 2.28 notes when we
get there.

aimeeu commented on content/en/armory-enterprise/release-notes/rn-armory-operator/oss-operator-v1-3-0.md:
```diff
@@ -0,0 +1,32 @@
+Operator now supports Kubernetes versions greater than 1.21. The following
+table outlines compatible versions of Operator, Kubernetes, and Spinnaker.
+
+| Kubernetes Version         | Spinnaker Operator Version     | Spinnaker Version                |
```
Um... You changed the column heading to Spinnaker Operator Version but the actual versions are the same as the Armory Operator versions

aimeeu commented on content/en/includes/armory-operator/operator-compat-matrix.md:
```diff
@@ -0,0 +1,6 @@
+| < 1.21                     | <= 1.6.x                       | <= 2.28.0                         |
+| >= 1.21                    | >= 1.7.x                       | All Supported Enterprise Versions |
+
+For guidance on upgrading the Operator, consult our {{< linkWithTitle "op-manage-operator.md" >}} guide.
```
```suggestion
For guidance on upgrading the Operator, consult the {{< linkWithTitle "op-manage-operator.md" >}} guide.
```

aimeeu commented on content/en/armory-enterprise/installation/armory-operator/op-quickstart.md:
```diff
@@ -14,6 +14,12 @@ The goal of this guide is to deploy Armory Enterprise with bare minimum configur
 
 {{% include "armory-operator/k8s-reqs.md" %}}
 
+Depending on your Kubernetes version you may need to adjust the following
```
```suggestion
Depending on your Kubernetes version, you may need to adjust the following
```

aimeeu commented on content/en/armory-enterprise/installation/armory-operator/op-quickstart.md:
```diff
@@ -14,6 +14,12 @@ The goal of this guide is to deploy Armory Enterprise with bare minimum configur
 
+Depending on your Kubernetes version you may need to adjust the following
+instructions to use a support Operator version. The following table outlines
+the supported combinations of Kubernetes, Operator, and Armory Enterprise.
```
```suggestion
the supported combinations of Kubernetes, Operator, and Armory Enterprise:
```

aimeeu commented on content/en/armory-enterprise/release-notes/rn-armory-operator/armory-operator-v1-7-0.md:
```diff
@@ -0,0 +1,28 @@
+
+Operator now supports Kubernetes versions greater than 1.21. The following
+table outlines compatible versions of Operator, Kubernetes, and Armory
+Enterprise.
```
```suggestion
Enterprise:
```

aimeeu reviewed (changes requested):
A few nits and a question on the open source operator versions in the table

dogonthehorizon committed:
chore(release): apply suggestions from code review

Co-authored-by: aimee <aimeeu.opensource@gmail.com>

aimeeu commented on content/en/armory-enterprise/feature-status/armory-enterprise-matrix/_index.md:
```diff
@@ -71,6 +71,11 @@ The [Armory Operator]({{< ref "armory-operator" >}}) and [Spinnaker Operator](ht
 
+Note that newer versions of the Armory Operator drop support for older
+Kubernetes versions. The following table outlines the supported combinations of
+Kubernetes, the Operator, and Armory Enterprise.
```
```suggestion
Kubernetes, the Operator, and Armory Enterprise:
```

aimeeu committed:
Update content/en/armory-enterprise/feature-status/armory-enterprise-matrix/_index.md

dogonthehorizon committed:
chore(release): remove OSS release notes

Putting this in another branch as we'll release the OSS operator in a
few days.
